apps/e2e/integration/reconciliation_order.spec.ts

```diff
@@ -1706,3 +1706,48 @@ testOrders("commit: applies delivery updates to customer order lines", async ({
 	await expect(lNotAffected).toHaveCount(1);
 	await lNotAffected.nth(0).getByRole("cell", { name: "Placed" }).waitFor();
 });
+
+testOrders("commit: navigates back to suppliers/orders after the order is finalized", async ({ page, supplierOrders }) => {
+	// Navigate to supplier orders and start reconciliation
+	await page.goto(`${baseURL}orders/suppliers/orders/`);
+
+	const table = page.getByRole("table");
+
+	await page.getByRole("button", { name: "Ordered", exact: true }).click();
+
+	// NOTE: using the first order (from the fixture) for the test
+	// (not really relevant for this test, but we want to make sure it's, in fact, an order row, not a header)
+	const { order } = supplierOrders[0];
+	await table
+		.getByRole("row")
+		.filter({ has: page.getByRole("cell", { name: order.supplier_name, exact: true }) })
+		// TODO: replace nth(0) with the following line after merging the entire reconciliation related work
+		// .filter({ has: page.getByRole("cell", { name: order.totalBooks.toString(), exact: true }) })
+		.nth(0)
+		.getByRole("checkbox")
+		.click();
+
+	await page.getByText("Reconcile").first().click();
+
+	const isbnInput = page.getByPlaceholder("Enter ISBN of delivered books");
+
+	// Scan one book (so that we can commit)
+	await isbnInput.fill(supplierOrders[0].lines[0].isbn);
+	await page.keyboard.press("Enter");
+	await table.getByRole("row").filter({ hasText: supplierOrders[0].lines[0].isbn }).waitFor();
+
+	// Compare and commit
+	await page.getByRole("button", { name: "Compare", exact: true }).click();
+	await page.getByRole("button", { name: "Commit", exact: true }).click();
+	const dialog = page.getByRole("dialog");
+	await dialog.getByRole("button", { name: "Confirm" }).click();
+	await dialog.waitFor({ state: "detached" });
+
+	// Should have navigated back to supplier orders
+	//
+	// These buttons should be enough to conclude we're back at the supplier orders page
+	await page.getByRole("button", { name: "Unordered", exact: true }).waitFor();
+	await page.getByRole("button", { name: "Ordered", exact: true }).waitFor();
+	await page.getByRole("button", { name: "Reconciling", exact: true }).waitFor();
+	await page.getByRole("button", { name: "Completed", exact: true }).waitFor();
+});
```

Comment:
This is the actual "meat" of the test: making sure we're redirected after confirming the reconciliation.

I'm thinking about pros and cons of this approach (a single test is responsible for a single piece of behaviour) vs adding assertions to existing tests.

My instinct goes to keeping the number of tests in check, so I'd prefer having this added as part of the existing three tests.

Two of them state `// Should navigate to supplier orders` and then proceed to click a UI link to navigate to a different page. I think replacing that comment with these checks would have the advantage of not introducing an additional test. And the disadvantage that a single test has too many responsibilities.

Reply:
Right...so, this was done as part of this work (to provide a test for the functionality). It could have been added to an existing test, but the existing tests were under refactor. Afterwards, the refactor came in and became part of `main` (and, by rebase, this branch).
I've moved this assertino to an existing test